crates/oxc_isolated_declarations/tests/fixtures/expando-function.ts

```diff
@@ -16,14 +16,19 @@ export namespace foo {
   export let baz = 100;
 }
 
-// namespace must be exported
-namespace foo {
-  export let bar = 42;
-}
```

Comment:
TS will report an error for `foo.bar = 42` below, but oxc won't report an error now. This is still actually a typecheck error, `Individual declarations in merged declaration 'foo' must be all exported or all local.` so I feel like not having an error in declaration emit is fair game. Arguably that typecheck error should be made into an isolated declaration error, but that's not something oxc needs to worry about until TS implements that.